app/helpers/exercise_input_helper.rb

```diff
@@ -15,6 +15,18 @@ def default_content_tag_id(exercise)
     exercise.custom? ? 'mu-custom-editor-default-value' : 'default_content'
   end
 
+  def render_exercise_read_only_editor(exercise, content)
+    render "layouts/exercise_inputs/read_only_editors/#{exercise.editor}", exercise: exercise, content: content
+  end
+
+  def input_form_for(exercise)
+    if exercise&.input_kids?
+      'kids'
+    else
+      exercise.class.name.underscore
+    end
+  end
+
   def should_render_problem_tabs?(exercise, user)
     !exercise.hidden? && (exercise.queriable? || exercise.extra_visible? || exercise.has_messages_for?(user))
```

Comment:
This code is similar to `should_render_exercise_tabs`, and my intuition says that there may be similar codes. Can we do something about it?